modules/chat/src/main/ChatUi.scala

```diff
@@ -64,34 +64,53 @@ final class ChatUi(helpers: Helpers):
       palantir: Boolean = false,
       hostIds: List[UserId] = Nil
   )(using ctx: Context): JsObject =
-    Json
-      .obj(
-        "data" -> Json
-          .obj(
-            "id"         -> chat.id,
-            "name"       -> name,
-            "lines"      -> lines,
-            "resourceId" -> resourceId.value
-          )
-          .add("hostIds" -> hostIds.some.filter(_.nonEmpty))
-          .add("userId" -> ctx.userId)
-          .add("loginRequired" -> chat.loginRequired)
-          .add("restricted" -> restricted)
-          .add("palantir" -> (palantir && ctx.isAuth)),
-        "writeable" -> writeable,
-        "public"    -> public,
-        "permissions" -> Json
-          .obj("local" -> (public && localMod))
-          .add("broadcast" -> (public && broadcastMod))
-          .add("timeout" -> (public && Granter.opt(_.ChatTimeout)))
-          .add("shadowban" -> (public && Granter.opt(_.Shadowban)))
-      )
-      .add("kobold" -> ctx.troll)
-      .add("blind" -> ctx.blind)
-      .add("timeout" -> timeout)
-      .add("noteId" -> (withNoteAge.isDefined && ctx.noBlind).option(chat.id.value.take(8)))
-      .add("noteAge" -> withNoteAge)
-      .add(
-        "timeoutReasons" -> (!localMod && (Granter.opt(_.ChatTimeout) || Granter.opt(_.BroadcastTimeout)))
-          .option(JsonView.timeoutReasons)
-      )
+    val noteId = (withNoteAge.isDefined && ctx.noBlind).option(chat.id.value.take(8))
+    if ctx.kid.yes then
+      Json
+        .obj(
+          "data" -> Json
+            .obj(
+              "id"         -> "",
+              "name"       -> "",
```

Comment:
no, this is not good. If the ts typing says that there exists `data.id: string` and `data.name: string`, and you pass them empty strings, then you just made it useless.

I don't want our TS code to check if every string is empty before use.

Reply:
I may have misunderstood your point in the first place.
Hopefully this fix will be what you had in mind 
Thanks for your patience and your feedback

Reply:
I can see some errors in the console when playing with the notes. 
But it's also happening on master.

` {"SyntaxError":"JSON.parse: unexpected end of data at line 1 column 1 of the JSON data","stack":""}`